### PR TITLE
Add private types on 4.x

### PR DESCRIPTION
**Pull request by smnandre**
(WIP but i open to gather early feedback / comments)

Objective: add missing types 

- [x] Private properties
- [x] Return type on private methods
- [ ] Argument types on private methods (?not sure)

Maybe "@internal" then

### CODE REVIEW

**Comment by smnandre** [comment]
Remaining non-typed arguments inherit their "non-types" from protected or public methods, not sure what to do there 

Example in `Profiler\Dumper\BaseDumper`

```php

    abstract protected function formatTime(Profile $profile, $percent): string;

    private function dumpProfile(Profile $profile, $prefix = '', $sibling = false): string

```

File: src/ExpressionParser.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ class ExpressionParser
     private array $unaryOperators;
     /** @var array<string, array{precedence: int, class: class-string<AbstractBinary>, associativity: self::OPERATOR_*}> */
     private array $binaryOperators;
-    private $readyNodes = [];
+    private array $readyNodes = [];
```
Comment (fabpot) [inline src/ExpressionParser.php]: Removed in 2acd9e8d9fcf2fa7907db8a6e1bcdd8feffcd91f

File: src/NodeVisitor/EscaperNodeVisitor.php [hunk 2]
```diff
@@ -182,7 +182,7 @@ private function isSafeFor(string $type, Node $expression, Environment $env): bo
         return \in_array($type, $safe) || \in_array('all', $safe);
     }
 
-    private function needEscaping()
+    private function needEscaping(): bool
```
Comment (fabpot) [inline src/NodeVisitor/EscaperNodeVisitor.php]:
```suggestion
    private function needEscaping(): string|false
```

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by stof** [comment]
Actually, types on all private APIs could be added in the 3.x branch

**Comment by smnandre** [comment]
So i should target 3.x for this one @stof right ? And then deal with `@internal` for 4.x ?

File: src/ExpressionParser.php (hunk 1 above)
Comment (smnandre) [reply src/ExpressionParser.php]: Thanks, rebased and removed

**Comment by smnandre** [comment]
(is there something like a "fabbot_ignore"? to prevent false-positive ? asking more generally than for this PR)

**Comment by fabpot** [comment]
Let's not bother with 3.x.

**Comment by smnandre** [comment]
So this seems ok for a first step.

Let's create another one for @internal function/methods if you're ok ?

```
    /**
     * Strips HTML and PHP tags from a string.
     *
     * @param string|null          $string
     * @param string[]|string|null $allowable_tags
     *
     * @internal
     */
    public static function striptags($string, $allowable_tags = null): string
    {
        return strip_tags($string ?? '', $allowable_tags);
    }
```

Because here there may be a "soft-conflict" between the signature (`string|null`) and the code (passing null to `strip_tags` is deprecated since 8.1)

**Comment by fabpot** [comment]
Thank you @smnandre.

**Comment by stof** [comment]
@smnandre there is no conflict in your code snippet. The code is not passing null to `strip_tags`

**Comment by smnandre** [comment]
Wow... let's just attribute it to today's emotions. ;)

**Pull request merged**